hax/helper/generate_sysconf.py

```diff
@@ -0,0 +1,255 @@
+        filename = f'm0d-{fid}'
+        contents = (f'MOTR_M0D_EP={ep}\n'
+                    f'MOTR_HA_EP={hax_ep}\n'
+                    f'MOTR_PROCESS_FID={fid}\n'
+                    'MOTR_CONF_XC=/etc/motr/confd.xc\n')
+        self._write_file(self.sysconfig_dir + filename, contents)
+
+    def generate_ios(self, svc_id: str, hax_ep: str):
+        fid = Fid(ObjT.PROCESS.value, int(svc_id))
+        ep = self.get_service_ep(svc_id)
+        meta_data = self.get_ios_meta_data(svc_id)
+        filename = f'm0d-{fid}'
+        contents = (f'MOTR_M0D_EP={ep}\n'
+                    f'MOTR_HA_EP={hax_ep}\n'
+                    f'MOTR_PROCESS_FID={fid}\n')
+        if meta_data:
+            contents += f'MOTR_BE_SEG_PATH={meta_data}\n'
+        self._write_file(self.sysconfig_dir + filename, contents)
+
+    def generate_s3(self, svc_id: str, hax_ep: str, s3_port: int):
+        profile_key = self.kv.kv_get('m0conf/profiles/', keys=True)
+        assert profile_key
+        # Get the first profile fid
+        profile_fid = profile_key[0].split('/')[-1]
+        fid = Fid(ObjT.PROCESS.value, int(svc_id))
+        ep = self.get_service_ep(svc_id)
+        filename = f's3server-{fid}'
+        contents = (f'MOTR_PROFILE_FID={profile_fid}\n'
+                    f'MOTR_S3SERVER_EP={ep}\n'
+                    f'MOTR_HA_EP={hax_ep}\n'
+                    f'MOTR_PROCESS_FID={fid}\n'
+                    f'MOTR_S3SERVER_PORT={s3_port}\n')
+        self._write_file(self.sysconfig_dir + filename, contents)
+
+    def generate_sysconfig(self):
+        s3_port = 28071
+        IDs = self.get_all_svc_ids()
+
+        hax_ep = self.get_service_ep(IDs['HAX_ID'][0])
+        for x in IDs['CONFD_IDs']:
+            self.generate_confd(x, hax_ep)
+
+        for x in IDs['IOS_IDs']:
+            self.generate_ios(x, hax_ep)
+
+        for x in IDs['S3_IDs']:
+            self.generate_s3(x, hax_ep, s3_port)
+            s3_port += 1
+
+    def append_svcs_to_file(self, svcs, conf_file):
+        with open(conf_file) as consul_conf_file:
+            data = simplejson.load(consul_conf_file)
+        data['services'] = svcs
+        self._write_file(conf_file,
+                         simplejson.dumps(data, indent=2, for_json=True))
+
+    def get_service_ipaddr(self, ep: str) -> str:
+        return ep.split('@')[0]
+
+    def append_ipaddr_to_file(self, hax_ep: str, conf_file: str):
+        ipaddr = self.get_service_ipaddr(hax_ep)
+        with fileinput.FileInput(conf_file, inplace=True,
+                                 backup='.bak') as file:
+            for line in file:
+                print(line.replace('http://localhost',
+                                   f'http://{ipaddr}'), end='')
+
+    def get_service_addr(self, ep: str) -> str:
+        return ep.rsplit('@', 1)[0]
+
+    def get_service_port(self, ep: str) -> int:
+        return int(ep.rsplit('@', 1)[1])
+
+    def prepare_svc(self, id: str, name: str):
```

Comment:
Just to make codacy happy, please rename the id to may be svc_id or fid?

Reply:
done